extension-manifest-v3/src/background/reporting.js

```diff
@@ -267,6 +270,20 @@ function onLocationChange(details) {
     }
 
     try {
+      if (pendingStartup) {
+        await Promise.race([
+          pendingStartup,
+          delay(2000).then(() => {
```

Comment:
The last thing, which I am not sure, is if we need that timeout. In this case (and adblocker too), we mainly waits for the local storage to return data, no another "real" async work. It should not take more than few ms. 
Another thing is that is anything goes wrong, the promise rejects and does not clear itself, so that case is also covered.

Reply:
If it rejects (or times out), it will be handled by the catch.

Reporting does more things during initialization. It's true that a slow startup (exceeding the timeout) would be a bug, but I would still leave the timeout. It's easy to accidentally introduce delay, for instance, if a network request hangs and blocks startup, etc.

A difference to the adblocker is also that queuing up request in reporting is arguably more harmful.